libs/base/Data/List.idr

```diff
@@ -1027,3 +1027,8 @@ export
 lengthReplicate : (n : Nat) -> length (replicate n x) = n
 lengthReplicate 0 = Refl
 lengthReplicate (S k) = cong S (lengthReplicate k)
+
+export
+foldlOfAppend : (xs : List a) -> (ys : List a) -> (f : acc -> a -> acc) -> (init : acc) -> foldl f init (xs ++ ys) = foldl f (foldl f init xs) ys
```

Comment:
Please use the same naming scheme as in the rest of the file:
* lemma of the shape `fun1 (fun2 xs) = ...` is named `fun1Fun2`
* no `Of` in the middle of the name
* arguments taken in the same left-to-right order they're in in the statement, here: `f, init, xs, ys

Reply:
I wonder why `++` is already called `append`. Isn't it a concatenation, when appending is what `snoc` function does?